src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,7 +223,20 @@ public static String formatDateTimeForComments(Date date) {
         if (date == null) {
             return "";
         }
-        SimpleDateFormat sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a zzz");
+        SimpleDateFormat sdf = null;
+        Calendar c = Calendar.getInstance();
+        c.setTime(date);
+        if (c.get(Calendar.HOUR_OF_DAY) == 12) {
+            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm");
+            sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return sdf.format(date) + " NOON UTC";
+        }
+        if (c.get(Calendar.HOUR_OF_DAY) == 0) {
+            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, KK:mm");
+            sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return sdf.format(date) + " MIDNIGHT UTC";
```

Comment:
After discussing this with @kanghj we decide that Midnight thing can be omitted. NOON is enough. The only time that can be confused is 12.00 PM. So just change `12.00 PM` to `12 NOON`. No other times need changing. For example 12.30 PM doesn't need changing.